### PR TITLE
Reorganize ihm filtre creation edition

### PR DESCRIPTION
**Pull request by Walid-Sahnoun**

### CODE REVIEW

File: src/components/dialogs/criteria-based-filter-dialog-content.js [hunk 1]
```diff
@@ -193,17 +209,76 @@ export const CriteriaBasedFilterDialogContent = ({
         currentFilterToSend.current.type = FilterType.CRITERIA;
         currentFilterToSend.current.equipmentFilterForm = { ...res };
         handleFilterCreation(currentFilterToSend.current);
+        const keys = Object.keys(res);
```
Comment (laurgarn) [inline src/components/dialogs/criteria-based-filter-dialog-content.js]: What about :
```
const hasEdition = Object.values().some((v) => (Array.isArray(v) && v.length > 0) || (typeof v === 'object' && Object.keys(v).length() > 0);
if (hasEdition) setIsEdited(true);
```
?

File: src/components/dialogs/criteria-based-filter-dialog-content.js [hunk 2]
```diff
@@ -193,17 +209,76 @@ export const CriteriaBasedFilterDialogContent = ({
...
+            ) {
+                setIsEdited(true);
+                return;
+            }
+        }
     };
 
-    const changeFilterType = (newType) => {
+    const changeEquipmentType = (newType) => {
         // TODO: should reset all fields in currentFormEdit
```
Comment (laurgarn) [inline src/components/dialogs/criteria-based-filter-dialog-content.js:223]: Is this comment still useful ?

File: src/components/dialogs/field-hook.js [hunk 3]
```diff
@@ -460,17 +460,19 @@ export const useEquipmentTableValues = ({
                 }
                 let objects = Object.keys(csvData).map(function (key) {
                     return {
-                        equipmentID: csvData[key][0].trim(),
+                        equipmentID: isGeneratorOrLoad
```
Comment (laurgarn) [inline src/components/dialogs/field-hook.js]: What justifies keeping side blanks when generator or load ?

File: src/components/dialogs/field-hook.js [hunk 4]
```diff
@@ -460,17 +460,19 @@ export const useEquipmentTableValues = ({
...
+                            ? csvData[key][0]
+                            : csvData[key][0]?.trim(),
                         distributionKey:
-                            csvData[key][1].trim() !== ''
+                            csvData[key][1]?.trim() !== ''
```
Comment (laurgarn) [inline src/components/dialogs/field-hook.js]: `csvData[key][1]?trim() || undefined` would do, no ?

File: src/components/dialogs/criteria-based-filter-dialog-content.js (hunk 1 above)
Comment (Walid-Sahnoun) [reply src/components/dialogs/criteria-based-filter-dialog-content.js]: yes, i changed like this

File: src/components/dialogs/criteria-based-filter-dialog-content.js (hunk 2 above)
Comment (Walid-Sahnoun) [reply src/components/dialogs/criteria-based-filter-dialog-content.js:223]: yes

File: src/components/dialogs/field-hook.js (hunk 3 above)
Comment (Walid-Sahnoun) [reply src/components/dialogs/field-hook.js]: yes, there is no difference, i changed

File: src/components/dialogs/field-hook.js (hunk 4 above)
Comment (Walid-Sahnoun) [reply src/components/dialogs/field-hook.js]: yes, i changed

File: src/components/dialogs/explicit-naming-filter-dialog-content.js [hunk 5]
```diff
@@ -162,7 +178,9 @@ const ExplicitNamingFilterDialogContent = ({
     const [isEdited, setIsEdited] = useState(false);
     const fetchFilter = useRef(null);
     fetchFilter.current = open && !isFilterCreation;
-
+    const intl = useIntl();
+    const [isConfirmationPopupOpen, setOpenConfirmationPopup] = useState(false);
+    const [newEquipmentType, setnewEquipmentType] = useState(null);
```
Comment (YenguiSeddik) [inline src/components/dialogs/explicit-naming-filter-dialog-content.js]: typo: set**N**ewEquipmentType**

Comment (Walid-Sahnoun) [reply src/components/dialogs/explicit-naming-filter-dialog-content.js]: ok

**Pull request closed without merging**